site/docs/v1/tech/developer-guide/api-gateways/haproxy-api-gateway.adoc

```diff
@@ -0,0 +1,185 @@
+* An HAProxy instance routed to an HTTP service. You can follow https://www.haproxy.com/blog/how-to-run-haproxy-with-docker/[this guide] to get a simple example running on Docker.
+
+[NOTE.note]
+====
+When creating your HAProxy container, be sure to use `haproxytech/haproxy-alpine:latest`, not `haproxytech/haproxy-alpine:2.4` as the guide instructs. Version 2.4 does not support the full set of JWT features needed in this tutorial.
+
+If you have trouble getting the `haproxy` container from the guide to start, make sure your `haproxy.cfg` file ends with a newline. Even whitespace on the last line will prevent the container from running.
+====
+
+
+== FusionAuth Configuration
+
+Navigate to your FusionAuth instance.
+
+First, you need to make sure that the JWT issuer setting is correct. Navigate to [breadcrumb]#Tenants -> Your Tenant# and change the [field]#Issuer# field to the URL of your FusionAuth instance, for example, `\https://local.fusionauth.io`. Record this value, because you will use it later when generating the JWT. It will be referred to as `<YOUR_FUSIONAUTH_URL>`.
+
+image::api-gateways/HAProxy-Api-Gateway/haproxy-tenant-issuer.png[Tenant issuer,width=1200px]
+
+Next, you need to configure a FusionAuth application to issue tokens to access the HAProxy services.
+
+Navigate to [breadcrumb]#Applications# and create a new application. Fill out the [field]#Name# field, then click the [breadcrumb]#JWT# tab and toggle the [field]#Enabled# switch. Select [uielement]#Auto generate a new key on save...# for the [field]#Access Token signing key# and [field]#Id token signing key# fields. This will generate an RS256 asymmetric key pair specifically for this application.
+
+image::api-gateways/HAProxy-Api-Gateway/haproxy-application-jwt-settings.png[Application JWT settings,width=1200px]
+
+Click the [uielement]#Save# button.
+
+After saving the new application, find it in the list of applications and click the [uielement]#View# button next to it (look for the green magnifying glass). Here you will find the application [field]#Id#. Record this value, as you will need it further on. It will be referred to later as `<YOUR_APPLICATION_ID>`.
+
+image::api-gateways/HAProxy-Api-Gateway/haproxy-application-id.png[Application id,width=1200px,role=bottom-cropped]
+
+Now, navigate to [breadcrumb]#Settings -> Key Master#. You will see the access key that you've just created in this list. By default, it has the name `Access token signing key generated for application <NAME_OF_APPLICATION>`. View the key and copy the entire contents of the [field]#PEM encoded# field under the [breadcrumb]#Public Key# section into a file in the same directory as you placed the `haproxy.cfg` file whilst setting up HAProxy in the quickstart guide. Name the file `pubkey.pem`. Remember this filename, as you will supply it to the HAProxy configuration file later.
```

Comment:
[nitpick (non-blocking):](https://conventionalcomments.org/) I think we could split _"View the key and ... in the quickstart guide"_ in two to make them simpler.

Reply:
done